homework/list.js

```diff
@@ -2,43 +2,51 @@
 
 (function() {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+	function listObject(){
+		this._listNoteContent = [];
+		this._wrapper = document.querySelector('#note-list-wrapper');
+	}
+  
+  listObject.prototype.start = function() {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
+    this.fetchList()
+    .then(this.updateList.bind(this))
+    .then(this.drawList.bind(this))
+    
+    .then(this.preloadFirstNote.bind(this));
+   
+    
     window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+      this.onNoteOpen(event);
+    }.bind(this));
   }
 
-  function onNoteOpen(event) {
+  listObject.prototype.onNoteOpen = function(event) {
     if (event.target.classList.contains('note-title')) {
       var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
+      var content = this._listNoteContent[id];
       window.dispatchEvent(new CustomEvent('note-open',
         { detail: content }));
     };
+    
   }
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
+  listObject.prototype.preloadFirstNote = function() {
+    if (this._listNoteContent.length !== 0) {
```

Comment:
If a function body is wrapped by a conditional statement with only one branch, it's a convention to move the `if...else` to the caller, rather than callee.